sdk/core/azure_core_macros/src/tracing_new.rs

```diff
@@ -219,35 +219,27 @@ pub fn parse_new(attr: TokenStream, item: TokenStream) -> Result<TokenStream> {
 fn is_arc_of_self(path: &syn::Path) -> std::result::Result<(), String> {
     let segment = path.segments.last().unwrap();
     if segment.ident != "Arc" {
-        warn!("Arc must be the first segment, found {:?}", segment.ident);
         return Err("Arc must be the first segment".to_string());
     }
     if segment.arguments.is_empty() {
-        warn!("Arc must have arguments, found {:?}", segment.arguments);
         return Err("Arc must have arguments".to_string());
     }
     match &segment.arguments {
         syn::PathArguments::AngleBracketed(AngleBracketedGenericArguments { args, .. }) => {
             if args.len() != 1 {
-                warn!("Arc must have one argument, found {args:?}");
                 return Err("Arc must have one argument".to_string());
```

Comment:
Per our offline discussion, I think it's fine to include more context in the errors. Just run it through `format!()`. If you want. Any information that's useful to you to fix macro problems I'm sure is useful to devs using them if it is actionable.